### ISSUE TITLE
model.predict() gives different results if asks to predict twice on the same test set, and converges to 1 in one of the categories

### ISSUE THREAD

**Issue by albertbuchard** [tflearn/tflearn#520]
Hi,

I do not know why and I have looked around without much results. I have a CNN like so 

```
# Real-time data preprocessing
img_prep = ImagePreprocessing()
img_prep.add_featurewise_zero_center()
img_prep.add_featurewise_stdnorm()

# Real-time data augmentation
img_aug = ImageAugmentation()
img_aug.add_random_flip_leftright()
img_aug.add_random_rotation(max_angle=90.)

# Convolutional network building
network = input_data(shape=[None, patch_size, patch_size, 3], name='input', 
                     data_preprocessing=img_prep,
                     data_augmentation=img_aug)
network = conv_2d(network, 32, 3, activation='relu', regularizer="L2")
network = max_pool_2d(network, 2)
network = conv_2d(network, 64, 3, activation='relu', regularizer="L2")
network = conv_2d(network, 64, 3, activation='relu')
network = max_pool_2d(network, 2)
network = fully_connected(network, 512, activation='relu')
network = dropout(network, 0.7)
network = fully_connected(network, 2, activation='softmax')

network = regression(network, optimizer='adam', learning_rate=0.001,
                     loss='categorical_crossentropy', name='target')

# Train using DNN
model = tflearn.DNN(network, tensorboard_verbose=3) 

# Fitting
model.fit({'input': X}, {'target':Y}, 
          n_epoch=1, shuffle=True, 
          validation_set=({'input':X_test}, {'target':Y_test}),
          show_metric=True, batch_size=92, run_id='current', snapshot_step=False)
```

And it converges to 

```
Training Step: 433  | total loss: 0.35395
| Adam | epoch: 001 | loss: 0.35395 - acc: 0.8356 | val_loss: 0.32121 - val_acc: 0.8545 -- iter: 39800/39800
```

But when I try to predict some of the training or validation set just to check, every time I predict, its prediction change and converges FAST to either one of the category, often the wrong one !

```
y_pre = model.predict(X[10:12])
print(y_pre)
print(Y[10:12])

# First one
>> [[0.710803210735321, 0.28919678926467896], [0.7572739720344543, 0.24272598326206207]]
>>[[ 0.  1.][ 0.  1.]]
>>(39800, 1) (2, 2)

# Second time with same data
>>[[0.9694352149963379, 0.030564822256565094], [1.0, 1.0934689598229852e-08]]
>>[[ 0.  1.] [ 0.  1.]]
>>(39800, 1) (2, 2)

# Third time with same data
>>[[0.9999997615814209, 2.0725028093693254e-07], [1.0, 0.0]]
>>[[ 0.  1.] [ 0.  1.]]
>>(39800, 1) (2, 2)
```

etc.
I dont understand this behavior, am I missing something ??

Thanks !

**Comment by scarletvv** [comment]
Hi,

I have the exactly same issue. Anyone has any thoughts about this issue?

Thanks!

**Comment by willduan** [comment]
mark

**Comment by scarletvv** [comment]
I may have found the reason. It is because we added the read-time data preprocessing in the model. Every time the model.predict is called, the data preprocessing is applied to the data set again and again. 

# Real-time data preprocessing
img_prep = ImagePreprocessing()
img_prep.add_featurewise_zero_center()
img_prep.add_featurewise_stdnorm()

**Comment by deepakanandece** [comment]
Any solution for this issue yet?

**Comment by hbarang** [comment]
I have the same problem, any solutions?
